File: v2/models/Customer.py
from pydantic import BaseModel, EmailStr, Field
from models.Account import Account
from datetime import datetime

class Customer(BaseModel):
    id: int # change to random generation
    username: str
    password: str

    name: str
    email: EmailStr
    accounts: list[Account] = Field(default_factory=list)

    time_created: datetime = Field(default_factory=datetime.now)

File: v4/frontend/src/types/customer.ts
export interface Customer {
    _id: string;
    username: string;
    name: string;
    email: string;
    time_created: string;
}

export interface LoginCredentials {
    username: string;
    password: string;
}

export interface RegisterCustomerPayload {
    username: string;
    password: string;
    name: string;
    email: string;
}